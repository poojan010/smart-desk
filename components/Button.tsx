import React from 'react';
import { Text, TouchableOpacity, GestureResponderEvent, StyleSheet } from 'react-native';

interface ButtonProps {
  text: string;
  onPress?: (event: GestureResponderEvent) => void;
}

const Button: React.FC<ButtonProps> = ({ text, onPress }) => (
  <TouchableOpacity testID='button' onPress={onPress} style={styles.button}>
    <Text style={styles.text}>{text}</Text>
  </TouchableOpacity>
);

const styles = StyleSheet.create({
  button: {
    padding: 10,
    backgroundColor: '#007bff',
    borderRadius: 5,
    alignItems: 'center',
  },
  text: {
    color: '#fff',
    fontSize: 16,
  },
});

export default Button;
