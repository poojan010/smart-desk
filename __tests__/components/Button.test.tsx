import { render, screen, userEvent, act } from "@testing-library/react-native";

import Button from "@/components/Button";


describe("<Button/>", () => {

    test("Snapshot", () => {
        const { toJSON } = render(<Button text="hello" />)

        expect(toJSON()).toMatchSnapshot()
    })

    test("Button Renders correctly", () => {
        render(<Button text="hello" />)

        screen.getByTestId("button")
        screen.getByText("hello")
    })

    test("Button onPress works correctly",async () => {
        const user = userEvent.setup();

        const onPressMock = jest.fn()

        render(<Button text="Submit" onPress={onPressMock} />);

        const button = screen.getByTestId("button")

        await act(async () => {
            await user.press(button);
        });

        expect(onPressMock).toHaveBeenCalled()
    })
})