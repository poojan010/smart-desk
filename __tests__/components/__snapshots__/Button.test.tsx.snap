// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> Snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#007bff",
      "borderRadius": 5,
      "opacity": 1,
      "padding": 10,
    }
  }
  testID="button"
>
  <Text
    style={
      {
        "color": "#fff",
        "fontSize": 16,
      }
    }
  >
    hello
  </Text>
</View>
`;
